test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,917 @@
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                $rs.Dispose()
+            }
+
+            $exception.FullyQualifiedErrorId | Should Match "DriveNotFoundException"
+        }
+    }
+
+    Describe "Get content syntax in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that the get content syntax returns null value in constrained language without provider" {
+
+            $iss = [initialsessionstate]::Create()
+            $iss.LanguageMode = "ConstrainedLanguage"
+            $rs = [runspacefactory]::CreateRunspace($iss)
+            $rs.Open()
+            $pl = $rs.CreatePipeline('${c:\temp\foo.txt}')
+
+            $result = $pl.Invoke()
+            $rs.Dispose()
+
+            $result[0] | Should Be $null
+        }
+    }
+
+    Describe "Stream redirection in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that stream redirection doesn't work in constrained language mode without provider" {
+
+            $iss = [initialsessionstate]::CreateDefault2()
+            $iss.Providers.Clear()
+            $iss.LanguageMode = "ConstrainedLanguage"
+            $rs = [runspacefactory]::CreateRunspace($iss)
+            $rs.Open()
+            $pl = $rs.CreatePipeline('"Hello" > c:\temp\foo.txt')
+
+            try
+            {
+                $pl.Invoke()
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                $rs.Dispose()
+            }
+
+            $exception.FullyQualifiedErrorId | Should Match "CmdletInvocationException"
+        }
+    }
+
+    Describe "Invoke-Expression in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        BeforeAll {
+
+            function VulnerableFunctionFromFullLanguage { Invoke-Expression $Args[0] }
+
+            $TestCasesIEX = @(
+                @{testName = "Verifies direct Invoke-Expression does not bypass constrained language mode"; 
+                  scriptblock = { Invoke-Expression '[object]::Equals("A", "B")' } }
+                @{testName = "Verifies indirect Invoke-Expression does not bypass constrained language mode"; 
+                  scriptblock = { VulnerableFunctionFromFullLanguage '[object]::Equals("A", "B")' } }
+            )
+        }
+
+        It "<testName>" -TestCases $TestCasesIEX {
+
+            param ($scriptblock)
+
+            try
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableConstrainedLanguageMode
+
+                & $scriptblock
+                throw 'No Exception!'
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode

```

Comment:
You repeat this enough I think it would be better to have it in an AfterEach

Reply:
See Test Note comment at top of file.  Pester AfterEach and AfterAll is not reliable when session is constrained language or locked down.

Reply:
Got it.  Thanks.